pyanaconda/kickstart.py

```diff
@@ -770,51 +770,58 @@ def parse(self, args):
 
         return fc
 
-class Firewall(commands.firewall.F28_Firewall):
+class Firewall(RemovedCommand):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.packages = []
 
+    def __str__(self):
+        # The kickstart for this command is generated by the Firewall sub module
+        return ""
+
     def setup(self):
-        if self.seen:
+        firewall_proxy = NETWORK.get_proxy(FIREWALL)
+        if firewall_proxy.FirewallKickstarted:
             self.packages = ["firewalld"]
 
     def execute(self, storage, ksdata, instClass):
         args = []
 
+        firewall_proxy = NETWORK.get_proxy(FIREWALL)
         # If --use-system-defaults was passed then the user wants
         # whatever was provided by the rpms or ostree to be the
         # default, do nothing.
-        if self.use_system_defaults:
+        if firewall_proxy.UseSystemDefaults:
             firewall_log.info("ks file instructs to use system defaults for "
                               "firewall, skipping configuration.")
             return
 
         # enabled is None if neither --enable or --disable is passed
         # default to enabled if nothing has been set.
-        if self.enabled == False:
+        if firewall_proxy.FirewallEnabled == False:
             args += ["--disabled"]
         else:
             args += ["--enabled"]
 
-        if "ssh" not in self.services and "ssh" not in self.remove_services and "22:tcp" not in self.ports:
```

Comment:
This is really unreadable and you have more than 99 characters on line here. Please abstract this somehow.

Reply:
Yeah, that really should look better.